openapi-codegen/core/src/main/scala/sttp/tapir/codegen/ClassDefinitionGenerator.scala

```diff
@@ -46,6 +60,8 @@ class ClassDefinitionGenerator {
         |object $name extends Enum[$name] with CirceEnum[$name] {
         |  val values = findValues
         |${indent(2)(members.mkString("\n"))}
+        |  implicit val ${name.head.toLower +: name.tail}Codec: sttp.tapir.Codec[List[String], ${name}, sttp.tapir.CodecFormat.TextPlain] =
+        |    makeQueryCodecForEnum(${name})
```

Comment:
**question:** Wouldn't that generate a codec in all enums, even if they are not query params?

Reply:
Yes... yes it would. It did cross my mind that it would be preferable to check whether an enum was actually being used as a query param before generating this serde... I can try to implement that additional check if you want

Reply:
edit: Ok I thought about it some more and I guess it's OK, since we can't know whether an enum will be used as a query param at this point.

Reply:
Or can we? :)

Reply:
We can

Reply:
I think it would be desirable, to save some compilation time for the generated code.

Reply:
Looks ok, thanks :)